src/ITensorMPS/alternating_update.jl

```diff
@@ -0,0 +1,147 @@
+using ITensors: permute
+using NDTensors: scalartype
+using Printf: @printf
+
+function _compute_nsweeps(t; time_step=default_time_step(t), nsweeps=default_nsweeps())
+  if isinf(t) && isnothing(nsweeps)
+    nsweeps = 1
+  elseif !isnothing(nsweeps) && time_step != t
+    error("Cannot specify both time_step and nsweeps in alternating_update")
+  elseif isfinite(time_step) && abs(time_step) > 0 && isnothing(nsweeps)
+    nsweeps = convert(Int, ceil(abs(t / time_step)))
+    if !(nsweeps * time_step ≈ t)
+      error("Time step $time_step not commensurate with total time t=$t")
+    end
+  end
+  return nsweeps
+end
+
+function _extend_sweeps_param(param, nsweeps)
+  if param isa Number
+    eparam = fill(param, nsweeps)
+  else
+    length(param) == nsweeps && return param
+    eparam = Vector(undef, nsweeps)
+    eparam[1:length(param)] = param
+    eparam[(length(param) + 1):end] .= param[end]
+  end
+  return eparam
+end
+
+function process_sweeps(; nsweeps, maxdim, mindim, cutoff, noise)
+  maxdim = _extend_sweeps_param(maxdim, nsweeps)
+  mindim = _extend_sweeps_param(mindim, nsweeps)
+  cutoff = _extend_sweeps_param(cutoff, nsweeps)
+  noise = _extend_sweeps_param(noise, nsweeps)
+  return (; maxdim, mindim, cutoff, noise)
+end
+
+function alternating_update(
+  solver,
+  PH,
+  t::Number,
+  psi0::MPS;
+  nsweeps=default_nsweeps(),
+  checkdone=default_checkdone(),
+  write_when_maxdim_exceeds=default_write_when_maxdim_exceeds(),
+  nsite=default_nsite(),
+  reverse_step=default_reverse_step(),
+  time_start=default_time_start(),
+  time_step=default_time_step(t),
+  order=default_order(),
+  (observer!)=default_observer!(),
+  (step_observer!)=default_step_observer!(),
+  outputlevel=default_outputlevel(),
+  normalize=default_normalize(),
+  maxdim=default_maxdim(),
+  mindim=default_mindim(),
+  cutoff=default_cutoff(Float64),
```

Comment:
Seems better to make this `default_cutoff(scalartype(psi0))`, say for example it is single precision then the cutoff should be adjusted. Same comment in other places where `default_cutoff` is used.

Of course in general you may want to account for the scalar type of the operator as well, but the state seems like a reasonable stand-in and we could always make it more sophisticated as needed.